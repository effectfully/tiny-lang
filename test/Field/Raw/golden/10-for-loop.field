for i = 1 to 2 do
    let i' = i;         
    for j = 2 to 3 do
        let k = i * j;
        assert k == i' * j;
    end;
    let p = i;
    for l = 1 to 2 do
        let p = p * l;
    end;      
end;                    
