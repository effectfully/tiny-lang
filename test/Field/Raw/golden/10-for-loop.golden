Program [] Statements [EFor (Var "i") 1 2 (Statements [ELet (Var "i'") (EVar (Var "i")),EFor (Var "j") 2 3 (Statements [ELet (Var "k") (EAppBinOp Mul (EVar (Var "i")) (EVar (Var "j"))),EAssert (EAppBinOp FEq (EVar (Var "k")) (EAppBinOp Mul (EVar (Var "i'")) (EVar (Var "j"))))]),ELet (Var "p") (EVar (Var "i")),EFor (Var "l") 1 2 (Statements [ELet (Var "p") (EAppBinOp Mul (EVar (Var "p")) (EVar (Var "l")))])])]